Makefile

```diff
@@ -0,0 +1,245 @@
+
```

Comment:
many of the commands here have mage alternative, we should consider replacing them in order to have better maintainability, this is not a blocker for this PR but I just want to document it:
test, clean, check, update, add-headers, and more.

Reply:
I agree. From what i see in other repos. Tools like MAKE or JUST are used as wrappers around mage commands and service other functions mage lacks. They should compliment each other and not overlap, We will need to organize this point.